src/constants.ts

```diff
@@ -86,3 +86,59 @@ export const MARKET_DEPTH_END_PRICE_SLIPPAGE_PERC = 20;
 
 // Logging
 export const NUMBER_SOURCES_PER_LOG_LINE = 12;
+
+// ABI for balance checker
+export const BALANCE_CHECKER_ABI = [
+    {
+        constant: true,
+        inputs: [
+            { name: 'user', type: 'address' },
+            { name: 'token', type: 'address' },
+        ],
+        name: 'tokenBalance',
+        outputs: [{ name: '', type: 'uint256' }],
+        payable: false,
+        stateMutability: 'view',
+        type: 'function',
+    },
+    {
+        constant: true,
+        inputs: [
+            { name: 'owners', type: 'address[]' },
+            { name: 'spenders', type: 'address[]' },
+            { name: 'tokens', type: 'address[]' },
+        ],
+        name: 'allowances',
+        outputs: [{ name: '', type: 'uint256[]' }],
+        payable: false,
+        stateMutability: 'view',
+        type: 'function',
+    },
+    {
+        constant: true,
+        inputs: [
+            { name: 'owner', type: 'address' },
+            { name: 'spender', type: 'address' },
+            { name: 'token', type: 'address' },
+        ],
+        name: 'allowance',
+        outputs: [{ name: '', type: 'uint256' }],
+        payable: false,
+        stateMutability: 'view',
+        type: 'function',
+    },
+    {
+        constant: true,
+        inputs: [
+            { name: 'users', type: 'address[]' },
+            { name: 'tokens', type: 'address[]' },
+        ],
+        name: 'balances',
+        outputs: [{ name: '', type: 'uint256[]' }],
+        payable: false,
+        stateMutability: 'view',
+        type: 'function',
+    },
+    { payable: true, stateMutability: 'payable', type: 'fallback' },
+];
+export const BALANCE_CHECKER_ADDRESS = '0x8d975E1a936BC9AaBB5d6cea9adCc5C3838070a7';
```

Comment:
Won't work outside of `mainnet`. This will fail on our Kovan deployment

Reply:
will work once we replace with asset-swapper tooling